File: CS/MainForm.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace CustomButtonsLayoutView
{
    public partial class MainForm : Form
    {
        public MainForm()
        {
            InitializeComponent();
        }

        private void MainForm_Load(object sender, EventArgs e)
        {
            var list = new BindingList<Customer>();
            for (int i = 0; i < 10; i++)
                list.Add(new Customer() { ID = i, Name = "Name" + i });
            myGridControlMain.DataSource = list;
        }

        private void layoutViewCustomButtonMain_CustomButtonClick(object sender, CustomButtonEventsArgs e)
        {
            MessageBox.Show(e.CustomButton.Name);
        }
    }
    public class Customer
    {
        public int ID { get; set; }
        public string Name { get; set; }
    }
}

File: VB/MainForm.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms

Namespace CustomButtonsLayoutView
	Partial Public Class MainForm
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub MainForm_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			Dim list = New BindingList(Of Customer)()
			For i As Integer = 0 To 9
				list.Add(New Customer() With {.ID = i, .Name = "Name" & i})
			Next i
			myGridControlMain.DataSource = list
		End Sub

		Private Sub layoutViewCustomButtonMain_CustomButtonClick(ByVal sender As Object, ByVal e As CustomButtonEventsArgs) Handles layoutViewCustomButtonMain.CustomButtonClick
			MessageBox.Show(e.CustomButton.Name)
		End Sub
	End Class
	Public Class Customer
		Private privateID As Integer
		Public Property ID() As Integer
			Get
				Return privateID
			End Get
			Set(ByVal value As Integer)
				privateID = value
			End Set
		End Property
		Private privateName As String
		Public Property Name() As String
			Get
				Return privateName
			End Get
			Set(ByVal value As String)
				privateName = value
			End Set
		End Property
	End Class
End Namespace
